program/src/instruction.rs

```diff
@@ -1,184 +1,420 @@
     /// Burns the given amount of tokens from the vault account, and reduces the
     /// amount in the stake account.
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[s]` Slash authority
-    /// 3. `[w]` Vault token account
-    /// 4. `[]` Vault authority, PDA with seeds `['token-owner', stake_config]`
-    /// 5. `[]` SPL Token program
-    ///
-    /// Instruction data: amount of tokens to slash
+    /// 
+    /// Instruction data: amount of tokens to slash.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        signer,
+        name = "slash_authority",
+        desc = "Stake account"
+    )]
+    #[account(
+        3,
+        writable,
+        name = "vault",
+        desc = "Vault token account"
+    )]
+    #[account(
+        4,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
+    #[account(
+        5,
+        name = "spl_token_program",
+        desc = "SPL Token program"
+    )]
     Slash(u64),
 
-    /// Sets new authority on a config or stake account
-    ///
-    /// 0. `[w]` Config or stake account
-    /// 1. `[s]` Current authority
-    /// 2. `[]` New authority
-    SetAuthority(Authority),
+    /// Sets new authority on a config or stake account.
+    #[account(
+        0,
+        writable,
+        name = "account",
+        desc = "Config or Stake config account"
+    )]
+    #[account(
+        1,
+        signer,
+        name = "authority",
+        desc = "Current authority on the account"
+    )]
+    #[account(
+        2,
+        name = "new_authority",
+        desc = "Authority to set"
+    )]
+    SetAuthority(AuthorityType),
 
-    /// Updates configuration parameters
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[s]` Config authority
+    /// Updates configuration parameters.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        signer,
+        name = "config_authority",
+        desc = "Stake config authority"
+    )]
     UpdateConfig(ConfigField),
 
-    /// Moves SOL rewards to the config and updates the stake rewards total
-    ///
-    /// Accounts expected by this instruction:
-    ///
-    /// 0. `[w,s]` Reward payer
-    /// 1. `[w]` Config account
-    /// 2. `[]` System Program
+    /// Moves SOL rewards to the config and updates the stake rewards total.
+    #[account(
+        0,
+        writable,
+        signer,
+        name = "payer",
+        desc = "Reward payer"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        2,
+        name = "system_program",
+        desc = "System program account"
```

Comment:
nit: we don't typically add the "account" suffix for programs
```suggestion
        desc = "System program"
```